### PR TITLE
Add IMDB dataset without need of NLTK

### PR DESCRIPTION
**Pull request by wangkuiyi**
Fixes https://github.com/PaddlePaddle/Paddle/issues/1491

This PR and https://github.com/PaddlePaddle/Paddle/pull/1456 both provide datasets for sentiment analysis. But seems both good.

### CODE REVIEW

File: python/paddle/v2/dataset/common.py [hunk 1]
```diff
@@ -32,3 +32,10 @@ def download(url, module_name, md5sum):
             shutil.copyfileobj(r.raw, f)
 
     return filename
+
+
+def dict_add(a_dict, ele):
```
Comment (helinwang) [inline python/paddle/v2/dataset/common.py]: Does `dict_add` need to be added `__all__` before other paddle paddle packages being able to use it?

Comment (wangkuiyi) [inline python/paddle/v2/dataset/common.py]: No. We don't need to do it.

I tried and it seems that adding to __all__ just helps `from package import *`.

File: python/paddle/v2/dataset/imdb.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        i += 1
+        doc = qs[i % 2].get()
+        while doc != None:
+            yield [word_idx.get(w, UNK) for w in doc], i % 2
+            doc = qs[i % 2].get()
+
+    return reader()
+
+
+def train(word_idx):
```
Comment (helinwang) [inline python/paddle/v2/dataset/imdb.py]: What is the benefit of allowing user to customize `word_idx`, comparing to:
```
word_idx = paddle.v2.dataset.imdb.build_dict(TRAIN_PATTERN,150)
def train():
    return reader_creator(
    re.compile("aclImdb/train/pos/.*\.txt$"),
    re.compile("aclImdb/train/neg/.*\.txt$"), word_idx, 1000)
```

Comment (wangkuiyi) [inline python/paddle/v2/dataset/imdb.py]: I hope that the user would explicitly call `build_dict` to build the dictionary in the userspace code, so we don't have to maintain a global variable in this package keeping the dictionary.  In short, I want to remove/prevent from global variables from PaddlePaddle's codebase.

**Pull request merged**